include/micm/configure/solver_config.hpp

```diff
@@ -361,51 +366,66 @@ namespace micm
       const std::string THIRD_BODY = "THIRD_BODY";
 
       ValidateSchema(object, { NAME, TYPE }, { TRACER_TYPE, ABS_TOLERANCE, DIFFUSION_COEFF, MOL_WEIGHT });
-
-      std::string name = object[NAME].get<std::string>();
+      
+      std::string name = object[NAME].as<std::string>();
       Species species{ name };
 
       // Load remaining keys as properties
-      for (auto& [key, value] : object.items())
+      for(auto it = object.begin(); it != object.end(); ++it)
       {
+        auto key = it->first.as<std::string>();
+        auto value = it->second;
+
         if (key != NAME && key != TYPE)
-        {
-          if (value.is_string())
+        {       
+          std::string stringValue = value.as<std::string>();
+          if (!stringValue.empty() && isInt(stringValue)) {
+            species.SetProperty<int>(key, value.as<int>());
+          }
+          else if (!stringValue.empty() && isFloat(stringValue)) {
+            species.SetProperty<double>(key, value.as<double>());
+          }
+          else if (!stringValue.empty() && isBool(stringValue))
+          {
+            species.SetProperty<bool>(key, value.as<bool>());
+          }
+          else
```

Comment:
It seems that YAML-cpp provides functionality to [check that type](https://github.com/jbeder/yaml-cpp/wiki/Tutorial). We should use that here, maybe? That would remove the need for these one-off type casting functions.

```
switch (node.Type()) {
  case YAML::NodeType::Null: // ...
  case YAML::NodeType::Scalar: // ...
  case YAML::NodeType::Sequence: // ...
  case YAML::NodeType::Map: // ...
  case YAML::NodeType::Undefined: 
}
```

Reply:
We cant use this node.Type as the code becomes very nested with lots of Try, catch statement and making it very complicated to read. 
Here is example what the code will look like. I dont think it pretty. 
if (node.IsScalar()) {
    // It's a single value (int, float, string, bool)
    std::cout << "This is a scalar value." << std::endl;

    // Check for specific types using as<>
    try {
        int intValue = node.as<int>();
        std::cout << "This is an integer: " << intValue << std::endl;
    } 
    catch (const YAML::BadConversion& e) {
        // Not an integer, try the next type
        try {
            double doubleValue = node.as<double>();
            std::cout << "This is a double: " << doubleValue << std::endl;
        } 
        catch (const YAML::BadConversion& e) {
            // Not a double, try string
            std::string stringValue = node.as<std::string>();
            std::cout << "This is a string: " << stringValue << std::endl;
        }
    }
} 

Thoughts?

Reply:
Ah, I didn't understand fully that yaml-cpp doesn't actually know the type. What you've done sounds fine and my comments no longer apply